synapse/lib/types.py

```diff
@@ -321,6 +359,71 @@ def norm(self, valu, oldval=None):
 
         return self._norm_list(valu)
 
+class XrefType(DataType):
+    '''
+    The XrefType allows linking a specific type of node to an inspecific
+    set of node forms.
+
+    Example Sub Type:
+
+        addType('foo:barrefs', subof='xref', source='bar,foo:bar')
+
+    '''
+
+    def __init__(self, tlib, name, **info):
+        DataType.__init__(self, tlib, name, **info)
+        self._sorc_type = None
+        self._sorc_name = None
+
+        sorc = info.get('source')
+
+        if sorc != None:
+            parts = sorc.split(',')
+            if len(parts) != 2:
+                raise BadInfoValu(name='source',valu=sorc,mesg='expected source=<name>,<type>')
+
+            self._sorc_name = parts[0]
+            self._sorc_type = parts[1]
+
+    def norm(self, valu, oldval=None):
+
+        if s_compat.isstr(valu):
+            return self._norm_str(valu, oldval=oldval)
+
+        if not islist(valu):
+            self._raiseBadValu(valu,mesg='Expected guid, psv, or list')
+
+        return self._norm_list(valu,oldval=None)
+
+    def _norm_str(self, text, oldval=None):
+
+        if len(text) == 32 and text.find('|') == -1:
+            return self.tlib.getTypeNorm('guid',text)
+
+        # FIXME full logical / quoted split
```

Comment:
going to table this for now ( we may have to design a new way for the syntax to do this since quoted fields will conflict with the syntax )